### PR TITLE
Add description fields to some fiat-crypto-adjacent packages

### PR DESCRIPTION
**Pull request by JasonGross**
Mostly lifted from the READMEs of the corresponding repos.

### CODE REVIEW

**Comment by JasonGross** [comment]
> I can suggest some compressed descriptions here if you want.

Yes, please do.

**Comment by gares** [comment]
It seems a copy paste of the project readme.md, there is a homepage field for giving the user quick access to that document

**Comment by palmskog** [comment]
@JasonGross are you fine with my suggested abbreviated descriptions? We can always improve later, but I think this is more manageable, and could also be used as part of the Coq Platform README.

**Comment by JasonGross** [comment]
> @JasonGross are you fine with my suggested abbreviated descriptions? We can always improve later, but I think this is more manageable, and could also be used as part of the Coq Platform README.

I don't see the suggestions, where are they?

File: extra-dev/packages/coq-bedrock2/coq-bedrock2.dev/opam [hunk 1]
```diff
@@ -20,6 +20,156 @@ depends: [
...
+-    list of instructions to initialize the stack pointer, call `init()`, call `loop()`, jump back to calling `loop()` followed by the compiled functions
+-    Meant to start execution at beginning of this list
+
+Correctness theorem: `compiler.CompilerInvariant.compiler_invariant_proofs`:
+-    There is an invariant `inv` on RISC-V machines, and
+     *    Here's how to establish `inv`
+     *    Running the machine for one step preserves `inv`
+     *   `inv` implies that the I/O trace of the machine is good
+- We call this the 'establish/preserve/use pattern'
+"""
```
[the remark this answers is not in the document]
Comment (JasonGross) [reply extra-dev/packages/coq-bedrock2/coq-bedrock2.dev/opam:172]: This is not quite right but I will commit this and then edit

File: extra-dev/packages/coq-fiat-crypto-with-bedrock/coq-fiat-crypto-with-bedrock.dev/opam [hunk 2]
```diff
@@ -28,6 +28,19 @@ conflict-class: [
...
+- [Jade Philipoom. Correct-by-Construction Finite Field Arithmetic in Coq. MIT Master's Thesis. February 2018.](http://adam.chlipala.net/theses/jadep_meng.pdf) Chapters 3 and 4 contain a detailed walkthrough of the field arithmetic implementations (again, targeting the previous compilation pipeline).
+- [Andres Erbsen. Crafting Certified Elliptic Curve Cryptography Implementations in Coq. MIT Master's Thesis. June 2017.](
+http://adam.chlipala.net/theses/andreser_meng.pdf) Section 3 contains a whirlwind introduction to synthesizing field arithmetic code using coq, without assuming Coq skills, but covering a tiny fraction of the overall library. Sections 5 and 6 contain the only write-up on the elliptic-curve library in this repository.
+- [Jason Gross. Performance Engineering of Proof-Based Software Systems at Scale. MIT Doctoral Thesis. February 2021.](http://adam.chlipala.net/theses/jgross.pdf) Chapters 4 and 5 describe the reflective program transformation framework at the center of the newest compilation pipeline.
+- The newest compilation pipeline as a whole does not have a separate document yet, but the README does go over it in some detail.
+"""
```
[the remark this answers is not in the document]
Comment (JasonGross) [reply extra-dev/packages/coq-fiat-crypto-with-bedrock/coq-fiat-crypto-with-bedrock.dev/opam:43]: C, bedrock2, Go, Rust, Zig, and experimentally Java and JSON.

File: extra-dev/packages/coq-bedrock2/coq-bedrock2.dev/opam [hunk 3]
```diff
@@ -20,6 +20,20 @@ depends: [
 ]
 dev-repo: "git+https://github.com/mit-plv/bedrock2.git"
 synopsis: "A work-in-progress language and compiler for verified low-level programming"
+description: """
+bedrock2 is a low-level systems programming language. One piece
+of the puzzle is a verified compiler targeting RISC-V. The source
+language itself is also equipped with a simple program logic for
+proving correctness of the source programs.
```
Comment (JasonGross) [inline extra-dev/packages/coq-bedrock2/coq-bedrock2.dev/opam:27]:
```suggestion
bedrock2 is a low-level systems programming language. This
language is equipped with a simple program logic for
proving correctness of the programs.  A verified compiler
targeting RISC-V from this language exists but is not yet on opam.
```

File: extra-dev/packages/coq-fiat-crypto-with-bedrock/coq-fiat-crypto-with-bedrock.dev/opam [hunk 4]
```diff
@@ -28,6 +28,10 @@ conflict-class: [
 ]
 dev-repo: "git+https://github.com/mit-plv/fiat-crypto.git"
 synopsis: "Cryptographic Primitive Code Generation by Fiat."
+description: """
+Coq code and proofs for a command-line binary that can synthesize proven-correct
+big-integer modular field arithmetic operations for cryptography.
```
Comment (JasonGross) [inline extra-dev/packages/coq-fiat-crypto-with-bedrock/coq-fiat-crypto-with-bedrock.dev/opam:33]:
```suggestion
big-integer modular field arithmetic operations for cryptography.
Target languages include C, Rust, Zig, Go, and bedrock2.
```

File: released/packages/coq-bedrock2/coq-bedrock2.0.0.1/opam [hunk 5]
```diff
@@ -20,6 +20,20 @@ depends: [
 ]
 dev-repo: "git+https://github.com/mit-plv/bedrock2.git"
 synopsis: "A work-in-progress language and compiler for verified low-level programming"
+description: """
+bedrock2 is a low-level systems programming language. One piece
+of the puzzle is a verified compiler targeting RISC-V. The source
+language itself is also equipped with a simple program logic for
+proving correctness of the source programs.
```
Comment (JasonGross) [inline released/packages/coq-bedrock2/coq-bedrock2.0.0.1/opam:27]:
```suggestion
bedrock2 is a low-level systems programming language. This
language is equipped with a simple program logic for
proving correctness of the programs.  A verified compiler
targeting RISC-V from this language exists but is not yet on opam.
```

File: released/packages/coq-fiat-crypto/coq-fiat-crypto.0.0.13/opam [hunk 6]
```diff
@@ -28,6 +28,10 @@ conflict-class: [
 ]
 dev-repo: "git+https://github.com/mit-plv/fiat-crypto.git"
 synopsis: "Cryptographic Primitive Code Generation by Fiat"
+description: """
+Coq code and proofs for a command-line binary that can synthesize proven-correct
+big-integer modular field arithmetic operations for cryptography.
```
Comment (JasonGross) [inline released/packages/coq-fiat-crypto/coq-fiat-crypto.0.0.13/opam:33]:
```suggestion
big-integer modular field arithmetic operations for cryptography.
Target languages include C, Rust, Zig, Go, and bedrock2.
```

File: extra-dev/packages/coq-fiat-crypto/coq-fiat-crypto.dev/opam [hunk 7]
```diff
@@ -28,6 +28,10 @@ conflict-class: [
 ]
 dev-repo: "git+https://github.com/mit-plv/fiat-crypto.git"
 synopsis: "Cryptographic Primitive Code Generation by Fiat."
+description: """
+Coq code and proofs for a command-line binary that can synthesize proven-correct
+big-integer modular field arithmetic operations for cryptography.
```
Comment (JasonGross) [inline extra-dev/packages/coq-fiat-crypto/coq-fiat-crypto.dev/opam:33]: Target languages include C, Rust, Zig, Go, and bedrock2.

Comment (JasonGross) [inline extra-dev/packages/coq-fiat-crypto/coq-fiat-crypto.dev/opam:33]:
```suggestion
big-integer modular field arithmetic operations for cryptography.
Target languages include C, Rust, Zig, Go, and bedrock2.
```

**Comment by JasonGross** [comment]
I've made a couple more edits, what do you think?

**Review by palmskog — APPROVED** [review]
Looks very nice now and informative for prospective users.

**Pull request merged**